azurerm/internal/services/network/application_gateway_resource.go

```diff
@@ -2232,6 +2239,9 @@ func expandApplicationGatewayHTTPListeners(d *schema.ResourceData, gatewayID str
 				Protocol:                    network.ApplicationGatewayProtocol(protocol),
 				RequireServerNameIndication: utils.Bool(requireSNI),
 				CustomErrorConfigurations:   customErrorConfigurations,
+				FirewallPolicy: &network.SubResource{
+					ID: utils.String(firewallPolicyID),
+				},
```

Comment:
I think this'll likely need to be conditionally set if `firewallPolicyID` is `!= ""` - or does this accept an empty string?